packages/react/src/wishlists/hooks/__tests__/useWishlist.test.tsx

```diff
@@ -20,129 +49,227 @@ jest.mock('react-redux', () => ({
-        ...mockWishlistState.wishlist,
-        error: mockError,
+
+    const {
+      result: {
+        current: { error },
       },
+    } = renderHook(() => useWishlist(), {
+      wrapper: withStore({
+        ...stateMockData,
+        wishlist: {
+          ...stateMockData.wishlist,
+          error: mockError,
+        },
+      }),
     });
 
     expect(error).toEqual(mockError);
   });
 
-  it('should render as empty', () => {
-    const { isEmpty } = getRenderedHook({
-      ...mockWishlistState,
-      wishlist: {
-        ...mockWishlistState.wishlist,
-        result: {
-          ...mockWishlistState.wishlist.result,
-          count: 0,
-          items: [],
-        },
-        items: {
-          ...mockWishlistState.wishlist.items,
-          ids: [],
+  it('should return correctly if empty', () => {
+    const {
+      result: {
+        current: {
+          isFetched,
+          data: { isEmpty },
         },
       },
+    } = renderHook(() => useWishlist(), {
+      wrapper: withStore({
+        ...stateMockData,
+        wishlist: {
+          ...stateMockData.wishlist,
+          result: {
+            ...stateMockData.wishlist.result,
+            count: 0,
+            items: [],
+          },
+          items: {
+            ...stateMockData.wishlist.items,
+            ids: [],
+          },
+        },
+      }),
     });
 
     expect(isEmpty).toBe(true);
+    expect(isFetched).toBe(true);
   });
 
-  it('should render in loading state due to the loading status', () => {
-    const { isLoading } = getRenderedHook({
-      ...mockWishlistState,
-      wishlist: {
-        ...mockWishlistState.wishlist,
-        isLoading: true,
+  it('should return loading state', () => {
+    const {
+      result: {
+        current: { isLoading },
       },
+    } = renderHook(() => useWishlist(), {
+      wrapper: withStore({
+        ...stateMockData,
+        wishlist: {
+          ...stateMockData.wishlist,
+          isLoading: true,
+        },
+      }),
     });
 
     expect(isLoading).toBe(true);
   });
 
-  it("should render in loading state while it doesn't begin fetching", () => {
-    const { isLoading } = getRenderedHook({
-      ...mockWishlistState,
-      wishlist: {
-        ...mockWishlistState.wishlist,
-        error: null,
-        result: undefined,
+  it('should not return data as loading if data was not fetched', () => {
+    const {
+      result: {
+        current: { isLoading },
       },
+    } = renderHook(() => useWishlist(), {
+      wrapper: withStore({
+        ...stateMockData,
+        wishlist: {
+          ...stateMockData.wishlist,
+          error: undefined,
+          result: {},
+        },
+      }),
     });
 
-    expect(isLoading).toBe(true);
+    expect(isLoading).toBe(false);
   });
 
   describe('actions', () => {
-    it('should call `addWishlistItem` action', () => {
-      const { addWishlistItem } = getRenderedHook();
+    it('should call `fetch` action', () => {
```

Comment:
What do you think of creating a specific describe block for testing the options of this hook? Something like this:

```
describe('options', () => {
  describe('enableAutoFetch', () => {
    it('should call `fetch` if it is true, () => {});

    it('should not call `fetch` if it is false, () => {});
  });

});

```